internal/k8s/reconciler/validator/gateway.go

```diff
@@ -0,0 +1,512 @@
+package validator
```

Comment:
This code was largely validation logic that was moved over from `reconciler/gateway.go` and `reconciler/listener.go`.